lib/spack/spack/solver/asp.py

```diff
@@ -1704,11 +1704,17 @@ def define_variant(
             pkg_fact(fn.variant_sticky(vid))
 
         # define defaults for this variant definition
+        default_values: Tuple[Union[bool, str], ...] = (variant_def.default,)
         if variant_def.multi:
-            for val in sorted(variant_def.make_default().values):
-                pkg_fact(fn.variant_default_value_from_package_py(vid, val))
-        else:
-            pkg_fact(fn.variant_default_value_from_package_py(vid, variant_def.default))
+            default_values = variant_def.make_default().values
+
+        for val in default_values:
+            pkg_fact(fn.variant_default_value_from_package_py(vid, val))
+
+        if variant_def.values is not None:
+            current_values = sorted(variant_def.values, key=lambda v: (v not in default_values, v))
```

Comment:
Why sorted? Using the provided order makes more sense to me.

Reply:
That's clear, but then for non-defaults you disrupt the order of possible values defined in `package.py`. E.g. `values=("c99", "c11", "c17")` or `values=("20", "17", "14", "11")` etc.

Reply:
I think it is. I'm getting the following on Python 3.14:

> With sorted: 0.847418 seconds
> With double loop: 0.735292 seconds

using

```python
if __name__ == "__main__":
    sorted_time = timeit.timeit(with_sorted, number=100000)
    double_loop_time = timeit.timeit(with_double_loop, number=100000)

    print(f"With sorted: {sorted_time:.6f} seconds")
    print(f"With double loop: {double_loop_time:.6f} seconds")
```

It also becomes more clear that there's an order, whereas with `sorted` you have to remember that Python's sort is a stable sort.